app/models/user.rb

```diff
@@ -5,6 +5,12 @@ class User < ApplicationRecord
   before_create :create_activation_digest
 
   has_many :microposts, dependent: :destroy
+  has_many :active_relationships, class_name: Relationship.name,
+    foreign_key: :follower_id, dependent: :destroy
+  has_many :passive_relationships, class_name: Relationship.name,
+    foreign_key: :followed_id, dependent: :destroy
+  has_many :following, through: :active_relationships, source: :followed
+  has_many :followers, through: :passive_relationships, source: :follower
```

Comment:
- Nếu quan hệ many-to-many có nhiều sắc thái, cái mà yêu cầu nhiều thực thể khác để diễn tả chúng, mình sử dụng has_many :through :source. Dùng class_name để chọn model sử dụng.